test/yellow/VaultUpgradability.spec.ts

```diff
@@ -0,0 +1,489 @@
+    });
+
+    it('admin can change admin', async () => {
+      await VaultImpl1.connect(implAdmin).changeAdmin(someone.address);
+      expect(await VaultImpl1.getAdmin()).to.be.equal(someone.address);
+    });
+
+    it('revert on someone changing admin', async () => {
+      await expect(VaultImpl1.connect(someone).changeAdmin(someother.address)).to.be.revertedWith(
+        NOT_ADMIN
+      );
+    });
+
+    // ======================
+    // ERC1822
+    // ======================
+    it('ERC1822 compliant', async () => {
+      const ImplementationSlot =
+        '0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc';
+      expect(await VaultImpl1.proxiableUUID()).to.be.equal(ImplementationSlot);
+    });
+
+    // ======================
+    // Newer implementation
+    // ======================
+    it('newer implementation is 0x0', async () => {
+      expect(await VaultImpl1.getNewerImplementation()).to.be.equal(AddressZero);
+    });
+
+    it('admin can set newer implementation', async () => {
+      expect(await VaultImpl1.getNewerImplementation()).to.be.equal(AddressZero);
+      const newerImplAddress = Wallet.createRandom().address;
+      await VaultImpl1.connect(implAdmin).setNewerImplementation(newerImplAddress);
+      expect(await VaultImpl1.getNewerImplementation()).to.be.equal(newerImplAddress);
+    });
+
+    it('revert on someone set newer implementation', async () => {
+      expect(await VaultImpl1.getNewerImplementation()).to.be.equal(AddressZero);
+      const newerImplAddress = Wallet.createRandom().address;
+      await expect(
+        VaultImpl1.connect(someone).setNewerImplementation(newerImplAddress)
+      ).to.be.revertedWith(NOT_ADMIN);
+    });
+
+    it('revert on setting different newer implementations twice', async () => {
+      expect(await VaultImpl1.getNewerImplementation()).to.be.equal(AddressZero);
+      const oneNewerImplAddress = Wallet.createRandom().address;
+      await VaultImpl1.connect(implAdmin).setNewerImplementation(oneNewerImplAddress);
+      expect(await VaultImpl1.getNewerImplementation()).to.be.equal(oneNewerImplAddress);
+
+      const otherNewerImplAddress = Wallet.createRandom().address;
+      await expect(
+        VaultImpl1.connect(implAdmin).setNewerImplementation(otherNewerImplAddress)
+      ).to.be.revertedWith(NEWER_IMPL_IS_SET);
+    });
+
+    it('revert on setting the same newer implementation twice', async () => {
+      expect(await VaultImpl1.getNewerImplementation()).to.be.equal(AddressZero);
+      const newerImplAddress = Wallet.createRandom().address;
+      await VaultImpl1.connect(implAdmin).setNewerImplementation(newerImplAddress);
+      expect(await VaultImpl1.getNewerImplementation()).to.be.equal(newerImplAddress);
+      await expect(
+        VaultImpl1.connect(implAdmin).setNewerImplementation(newerImplAddress)
+      ).to.be.revertedWith(NEWER_IMPL_IS_SET);
+    });
```

Comment:
Again, these tests are duplicates,

the test case here should be to be sure that an address can not be set twice, we don't care if it's the same or a different one.